### PR TITLE
[tune] Example for using huggingface hyperparamer_search API

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

This is the same as the `pbt_transformers` example, but using the official huggingface hyperparameter_search API with Ray as the backend.

WIP as we wait for the next release of transformers.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by richardliaw — COMMENTED** [review]
we can just merge it and depend on the master branch? (or a commit of the master branch)?

Also, can you deprecate the existing example then?

**Comment by amogkam** [comment]
+1 I think we should just replace the existing pbt_transformers example with this.

**Comment by amogkam** [comment]
@krfricke could you keep the name as pbt_transformer.py? That way any links to the code example arent broken.

**Comment by krfricke** [comment]
Absolutely. Done!

File: python/ray/tune/examples/pbt_transformers/pbt_transformers.py [hunk 1]
```diff
@@ -119,34 +32,76 @@ def tune_transformer(num_samples=8,
...
+    trainer = Trainer(
+        model_init=get_model,
+        args=training_args,
+        train_dataset=train_dataset,
+        eval_dataset=eval_dataset,
+        compute_metrics=build_compute_metrics_fn(task_name))
+
+    tune_config = {
+        "per_device_eval_batch_size": 32,
+        "per_device_train_batch_size": tune.choice([16, 32, 64]),
```
Comment (amogkam) [inline python/ray/tune/examples/pbt_transformers/pbt_transformers.py:96]: @krfricke I think you can actually remove `per_device_train_batch_size`,  `weight_decay`, and `learning_rate` from the config, so theres no need to specify it in both config and in PBT hyperparam_mutations. If these are not specified in config, the search space in PBT hyperparam_mutations is used instead.

**Review by richardliaw — APPROVED** [review]

**Pull request merged**